[RN][Blog] Add blogpost for 0.75 (#4176)
facebook/react-native-website: website/blog/2024-08-12-release-0.75.md

So you can still use `react-native init` today to create new projects using the Community CLI & template today, but you’ll be seeing a warning as the follows:

![Init Deprecation](../static/blog/assets/0.75-deprecation.png)

Starting from December 31st 2024, the init command won’t create a project anymore, and you’ll have to either:

- Use a framework such as Expo, with its own dedicate command to create a new project such as `npx create-expo-app`
- Invoke the Community CLI directly with `npx @react-native-community/cli init`

Please note that `react-native config` and all the other commands than `init` will continue working as usual.

:::info
In order to offer a smoother migration experience, the `react-native@0.75.0` package is still depending on `@react-native-community/cli` but we’re planning on removing this dependency in a future version of React Native.
:::

#### Auto-linking performance improvements

During this work in updating the `init` command, we also spent time rewriting the auto-linking logic to be more performant. This results in faster build speed for both Android and iOS.

> With React Native 0.75, the auto-linking step is now running twice as fast on iOS and ~9 times faster on Android. You can read more about these improvements in this dedicated post [here](https://github.com/react-native-community/discussions-and-proposals/discussions)

Reviewer: ```suggestion
With React Native 0.75, if you are using Expo, the auto-linking step could now be running ~1.5 as fast on iOS and ~6.5 times faster on Android. You can read more about these improvements in this dedicated post [here](https://github.com/react-native-community/discussions-and-proposals/discussions/814)
```

trying to average the faster time from my benchmark.
(1.35 + 1.64) / 2 ~= 1.5
(2.88 + 10.24) / 2 ~= 6.5

<!-- TODO: Draft here but link needs to be updated: Autolinking in 0.75 blogpost draft -->

### Yoga 3.1 and Layout Improvements

Since we last shipped Yoga [version 3.0](https://reactnative.dev/blog/2024/04/22/release-0.74#yoga-30) in React Native 0.74, we kept on pushing many improvements and new layout capabilities for your applications. React Native 0.75 ships with Yoga 3.1 and you can learn more about what’s new in it in the official Yoga’s [release blog post](https://www.yogalayout.dev/blog/announcing-yoga-3.1).

One notable and highly requested feature is the support for % in various places, such as `gaps` and `translation`

:::warning
These features are available only for the New Architecture. If you are keen to use them, please consider migrating to it.
:::

#### % in Gaps

> With 0.75, the `gap`, `clomumnGap` and `rowGap`props described [here](https://reactnative.dev/docs/flexbox#row-gap-column-gap-and-gap) now support a string with a % as possible value.

Reviewer 2: ```suggestion
With 0.75, the `gap`, `columnGap` and `rowGap`props described [here](https://reactnative.dev/docs/flexbox#row-gap-column-gap-and-gap) now support a string with a % as possible value.
```

In 0.75, we are removing the remaining methods that were in core:

- `UIManagerModule.showPopupMenu()`
- `UIManagerModule.dismissPopupMenu()`

As alternative, please use the please use the `<PopupMenuAndroid />` component which lives in the [`@react-native/popup-menu-android`](https://www.npmjs.com/search?q=%40react-native%2Fpopup-menu-android) package.

#### iOS: Finalized PushNotificationIOS deprecation work

In 0.74, we [deprecated](https://reactnative.dev/blog/2024/04/22/release-0.74#api-changes-to-pushnotificationios-deprecated) some APIs from the `PushNotificationIOS` module.

In 0.75, we’ve deleted these APIs to migrate off legacy representations of notifications metadata.

The APIs that have been deleted are:

```objc
  + (void)didReceiveLocalNotification:(UILocalNotification *)notification;
  + (void)didReceiveRemoteNotification:(NSDictionary *)notification;
```

Instead, use `didReceiveNotification:(UNNotification *)notification`.

#### Community CLI: Removal of ram-bundle and profile-hermes commands

> We would like two significant removals coming from the Community CLI: the commands `ram-bundle` and `profile-hermes` have been removed.

Reviewer 3: ```suggestion
We would like to announce two significant removals coming from the Community CLI: the commands `ram-bundle` and `profile-hermes` have been removed.
```

Not sure if that is the intended wording, but something is definitely missing in this sentence.

Reviewer 4: yup thank you

- [Android: PopUp Menu removed from core](/blog/2024/08/12/release-0.75#android-popup-menu-moved-to-separate-package)
- [iOS: Finalized Push Notifications deprecation work](/blog/2024/08/12/release-0.75#ios-finalized-pushnotificationios-deprecation-work)
- [Community CLI: Removal of ram-bundle and profile-hermes commands](/blog/2024/08/12/release-0.75#community-cli-removal-of-ram-bundle-and-profile-hermes-commands)

<!--truncate-->

## Highlights

### Yoga 3.1 and Layout Improvements

Since we last shipped Yoga [version 3.0](https://reactnative.dev/blog/2024/04/22/release-0.74#yoga-30) in React Native 0.74, we kept on pushing many improvements and new layout capabilities for your applications. React Native 0.75 ships with Yoga 3.1 and you can learn more about what’s new in the official Yoga’s [release blog post](https://www.yogalayout.dev/blog/announcing-yoga-3.1).

One notable and highly requested feature is the support for `%` values in various places, such as `gaps` and `translation`

:::info
These features are available only for the New Architecture. If you are keen to use them, please consider migrating to it.
:::

> #### `%` Values in Gaps

Reviewer 5: ```suggestion
#### Percentage Values in Gaps
```

Percent character looks a bit off at the beginning of headers, maybe we should use the word-based alternative?

> With 0.75, the `gap`, `columnGap` and `rowGap`props described [here](https://reactnative.dev/docs/flexbox#row-gap-column-gap-and-gap) now support a string with a `%` value as possible value.

Reviewer 6: missing a space

            width: 100,
            height: 100,
          }}
        />
        <View
          style={{
            backgroundColor: 'orange',
            width: 100,
            height: 100,
          }}
        />
      </View>
      <View
        style={{flex: 1, flexDirection: 'row', columnGap: '10%'}}>
        <View
          style={{
            backgroundColor: 'red',
            width: 100,
            height: 100,
          }}
        />
        <View
          style={{
            backgroundColor: 'violet',
            width: 100,
            height: 100,
          }}
        />
        <View
          style={{
            backgroundColor: 'magenta',
            width: 100,
            height: 100,
          }}
        />
      </View>
    </SafeAreaView>
  );
}
```

Will be rendered as follows:

| Android                                                      | iOS                                                  |
| ------------------------------------------------------------ | ---------------------------------------------------- |
| ![Android Gaps](../static/blog/assets/0.75-android-gaps.png) | ![iOS Gaps](../static/blog/assets/0.75-ios-gaps.png) |

> #### `%` values in Translation

Reviewer 5: ```suggestion
#### Percentage Values in Translation
```

Same case as above.
